axum-extra/src/multipart_builder.rs

```diff
@@ -0,0 +1,337 @@
+        // // the mime type (not strictly required by the spec, but always sent here)
+        // Content-Type: mime/type\r\n
+        // // if the part does not conform to the rest of the request's encoding,
+        // // this is specified
+        // Content-Transfer-Encoding: "ENCODING"\r\n
+        // // a blank line, then the contents of the file start
+        // \r\n
+        // CONTENTS\r\n
+
+        // Format what we can as a string, then handle the rest at a byte level
+        let mut serialized_part = format!("Content-Disposition: form-data; name=\"{}\"", self.name);
+        // specify a filename if one was set
+        if let Some(filename) = &self.filename {
+            serialized_part += &format!("; filename=\"{}\"", filename);
+        }
+        serialized_part += "\r\n";
+        // specify the MIME type
+        serialized_part += &format!("Content-Type: {}\r\n", self.mime_type);
+        // if an encoding was set, add that
+        // determine what encoding to label the body of the field with
+        let encoding: Option<&str> = match self.encoding {
+            TransferEncoding::Default => None,
+            TransferEncoding::Binary => Some("binary"),
+        };
+        if let Some(encoding) = encoding {
+            serialized_part += &format!("Content-Transfer-Encoding: {}\r\n", encoding);
+        }
+        serialized_part += "\r\n";
+        let mut part_bytes = serialized_part.as_bytes().to_vec();
+        part_bytes.extend_from_slice(&self.contents);
+        part_bytes.extend_from_slice(b"\r\n");
+
+        part_bytes
+    }
+}
+
+/// A boundary is defined as a user defined (arbitrary) value that does not occur in any of the data.
+/// Because the specification does not clearly define a methodology for generating boundaries, this implementation
+/// follow's Reqwest's, and generates a boundary in the format of `XXXXXXXX-XXXXXXXX-XXXXXXXX-XXXXXXXX` where `XXXXXXXX`
+/// is a hexadecimal representation of a pseudo randomly generated u64.
+fn generate_boundary() -> String {
+    let a = gen_boundary_portion();
+    let b = gen_boundary_portion();
+    let c = gen_boundary_portion();
+    let d = gen_boundary_portion();
+    format!("{a:016x}-{b:016x}-{c:016x}-{d:016x}")
+}
+
+/// Generate a single portion of the boundary by performing an xor
+/// shift on the current time
+fn gen_boundary_portion() -> u64 {
+    // *very* rudimentary xor-shift used
+    // the current time is used as a seed because it doesn't really matter what the data is, it should
+    // just be vaguely random
+    let mut x = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_millis() as u64;
+    x = x ^ (x << 13);
+    x = x ^ (x >> 17);
+    x = x ^ (x << 5);
+    x
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{MultipartForm, Part};
+    use axum::{body::Body, http};
+    use axum::{routing::get, Router};
+    use http::{Request, Response};
+    use http_body_util::BodyExt;
+    // for `collect`
```

Comment:
You don't need to write why you import something. Personally, I would recommend against it because it's just a matter of time before someone needs to change something here and suddenly the comments are wrong.